### PR TITLE
Student assignments

### PR DESCRIPTION
**Pull request by dadams22**

### CODE REVIEW

File: src/scenes/Assignment.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        },
+        {
+            label: 'idk mayne',
+            extension: 'image',
+            type: 'Image'
+        }]
+    }
+    const [errors, setErrors] = useState(Array(assignment.files.length).fill(''))
+    const dueDate = assignment.dueDate ? assignment.dueDate.format('MMMM DD') : null
+    const dueTime = assignment.dueDate ? assignment.dueDate.format('LT') : null
```
Comment (emmettmcdow) [inline src/scenes/Assignment.tsx:26]: is this supposed to be '.dueDate'?

File: src/scenes/Assignment.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    const tableSpace = {
+        width: '3em'
+    }
+
+    const checkFileType = (info: any, i: any) => {
+        return;
+        // does not work until file extensions are fixed
+        /*
+        let extension = info.file.name.split('.').pop();
+        let temp = errors.slice();
```
Comment (emmettmcdow) [inline src/scenes/Assignment.tsx:36]: This looks good to me. What's wrong with it?

File: src/scenes/Assignment.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        <Divider orientation='left' />
+        <Space direction='horizontal' size='large'>
+            <div><b>Due:</b> {dueDate} at {dueTime}</div>
+            <div><b>Points:</b> {assignment.points}</div>
+            <div><b>Required Files:</b> {assignment.files.length}</div>
+            {assignment.badge ? <div><b>Badge:</b> {assignment.badge}</div> : <></>}
+        </Space>
+        <Divider />
+        <div dangerouslySetInnerHTML={{__html: assignment.description}}></div>
+        <table>
```
Comment (emmettmcdow) [inline src/scenes/Assignment.tsx:76]: We might want to do this with antd layouts/rows/cols as opposed to table

File: src/scenes/Assignments.tsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    const dueTimes = assignments.map((assignment: any) => assignment.dueDate ? assignment.dueDate.format('LT') : null)
+    const [hover, setHover] = useState(-1);
+    const nonhoverStyle = {
+        borderTop: '1px solid black',
+    }
+    const hoverStyle = {
+        borderTop: '1px solid black',
+        backgroundColor: '#DBF3FA',
+    }
+    const hoverBottomStyle = {
```
Comment (emmettmcdow) [inline src/scenes/Assignments.tsx:124]: Might wanna move this out of the .tsx.

File: src/scenes/CreateAssignment.tsx [hunk 5]
```diff
@@ -0,0 +1,174 @@
+import React, { useState } from 'react';
```
Comment (emmettmcdow) [inline src/scenes/CreateAssignment.tsx:1]: Dis my file hehe

**Review by emmettmcdow — COMMENTED** [review]
Great job!

File: src/scenes/Assignment.tsx (hunk 3 above)
Comment (davidsonpoole) [reply src/scenes/Assignment.tsx:76]: I agree with Emmett, we should use some sort of UI framework for this. The logic looks good though. We will need to edit it a little bit to fit the assignment model type before we merge into development though

**Review by davidsonpoole — CHANGES_REQUESTED** [review]
Need to do some adjusting before merging into development, but looks good overall

**Comment by emmettmcdow** [comment]
All good!

**Pull request closed without merging**